app/src/main/java/com/squirtles/musicroad/map/MarkerIconView.kt

```diff
@@ -0,0 +1,87 @@
+package com.squirtles.musicroad.map
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.graphics.Rect
+import android.util.AttributeSet
+import android.util.Log
+import android.view.View
+import androidx.annotation.ColorInt
+import coil3.SingletonImageLoader
+import coil3.request.ImageRequest
+import coil3.request.allowHardware
+import coil3.request.transformations
+import coil3.toBitmap
+import coil3.transform.CircleCropTransformation
+
+class MarkerIconView(
+    context: Context,
+    attrs: AttributeSet? = null
+) : View(context, attrs) {
+    private val fillPaint = Paint().apply {
+        style = Paint.Style.FILL
+    }
+    private val strokePaint = Paint().apply {
+        style = Paint.Style.STROKE
+        strokeWidth = STROKE_WIDTH
+    }
+    private val imageLoader = SingletonImageLoader.get(context)
+    private var bitmap: Bitmap? = null
+    private val bitmapRect = Rect()
```

Comment:
onDraw() 함수 내에서만 사용되는 변수로 보이는데 클래스 변수로 두어야할 필요가 있을까요?

Reply:
`onDraw()` 안에서 객체 생성을 하면 아래와 같은 경고가 생깁니다.

![image](https://github.com/user-attachments/assets/01cad4e2-8cde-4bcb-b485-415035a1710b)

경고와 더불어 [Optimized a custom view](https://developer.android.com/develop/ui/views/layout/custom-views/optimizing-view?hl=en)에도 `In particular, eliminate allocations in onDraw(), because allocations might lead to a garbage collection that causes a stutter. Allocate objects during initialization or between animations.`라는 말이 있습니다. 그리기 작업을 하는 onDraw() 안에서 객체 생성을 하게 되면 가비지 컬렉션으로 인한 버벅거림이 발생할 수 있어서 onDraw()에서 객체 생성을 제거하는 것이 좋다고 합니다. 이러한 이유에서 Rect 객체를 미리 생성해뒀습니다.

> MarkerIconView 자동 정렬 수행하여 반영했습니다! 감사합니다